packages/xs-vat-worker/src/netstring.js

```diff
@@ -0,0 +1,51 @@
+export async function readNetstring(input) {
+  let prefix = Buffer.from([]);
+  let colonPos = -1;
+
+  const nextChunk = () =>
+    new Promise((resolve, _reject) => {
+      const rx = data => {
+        input.pause();
+        input.removeListener('data', rx);
+        resolve(data);
+      };
+      input.on('data', rx);
```

Comment:
is there an `input.once`? would it help?

Reply:
oops... the whole `netstring.js` is obsolete; it was part of the nodejs worker before integration with SwingSet.